src/c-writer.cc

```diff
@@ -1318,19 +1349,35 @@ void CWriter::WriteModuleInstance() {
 
 void CWriter::WriteGlobals() {
   Index global_index = 0;
-  if (module_->globals.size() != module_->num_global_imports) {
-    for (const Global* global : module_->globals) {
+  for (const Global* global : module_->globals) {
+    if (options_.no_sandbox && global->name == kStackPointerName) {
+      // Remember the global to write it later (it needs to be part of .c file)
+      stack_pointer_global_ = global;
+    } else {
       bool is_import = global_index < module_->num_global_imports;
       if (!is_import) {
         WriteGlobal(*global, DefineGlobalScopeName(ModuleFieldType::Global,
                                                    global->name));
         Write(Newline());
       }
-      ++global_index;
     }
+    ++global_index;
   }
 }
 
+void CWriter::WriteStackPointerGlobal(const Global& global) {
+  assert(options_.no_sandbox);
+  // Ideally we would want to have a guard page here in order to
+  // catch stack-overflows. For now just add 4k as a buffer page.
+  // We also add a range check against kStackSize on GlobalSet,
+  // note that since leaf functions do not set the __stack_pointer
+  // they won't trigger the range check in global.set
+  Write("__thread uint8_t ", kStackArrayVariableName, "[", kStackSize,
```

Comment:
You can use the new WASM_RT_THREAD_LOCAL macro here maybe?  It landed in https://github.com/WebAssembly/wabt/pull/2126

Reply:
done.